### ISSUE TITLE
JVM Crash when loading CRT

### ISSUE THREAD

**Issue by jpg0** [aws/aws-iot-device-sdk-java-v2#600]
### Describe the bug

When my java code attempts to load anything that depends on CRT, the JVM crashes when it's loaded.

### Expected Behavior

No crash

### Current Behavior

JVM crashes and (sometimes) produces a core dump and error log.

### Reproduction Steps

Currently this only happens in quite a large application, teasing it apart into a single case isn't simple. The same code appears ok when run standalone.

### Additional Information/Context

There is loads of data in the error log:

[hs_err_pid25.log](https://github.com/user-attachments/files/17441933/hs_err_pid25.log)

### SDK version used

1.21.0

### Environment details (OS name and version, etc.)

Ubuntu 22.04 docker host, Debian guest; aarch64

**Comment by jmklix** [comment]
Can you provide more details about this crash and your environment:
- What java version are you using?
- are you installing this sdk from source or mvn?
- a docker container that reproduces your environment
- any other details that might be helpful for debugging.

**Comment by jpg0** [comment]
I ended up building the CRT myself with debug symbols, and it would crash due to a missing GLIBC 2.32 dependency. I realised that I was using Debian 11 which ships GLIBC 2.31, so switched to Debian 12. This has ultimately fixed the problem for me, however it did require quite a lot of work on my behalf to figure out the problem. It would be good if there were some message in the logs etc that explained the problem, rather than just a hard crash. Regardless, I'll close this issue now.
